numbat/src/vm.rs

```diff
@@ -715,19 +715,18 @@ impl Vm {
                     let base = rhs.to_base_unit_representation();
                     let seconds_f = base.unsafe_value().to_f64();
 
-                    let duration = chrono::Duration::try_seconds(seconds_f.trunc() as i64)
-                        .ok_or(RuntimeError::DurationOutOfRange)?
-                        + chrono::Duration::nanoseconds(
-                            (seconds_f.fract() * 1_000_000_000f64).round() as i64,
-                        );
+                    let span = jiff::Span::new()
+                        .try_seconds(seconds_f.trunc() as i64)
+                        .map_err(|_| RuntimeError::DurationOutOfRange)?
+                        .nanoseconds((seconds_f.fract() * 1_000_000_000f64).round() as i64);
```

Comment:
This is still cumbersome. It would be great if I could create a `jiff::Span` from a `f64` value in seconds.

Reply:
If you know your `seconds_f` is always positive, you can create a `std::time::Duration` from it and then use `Span::try_from(duration)`.

If it's possibly negative, my plan is to add a `jiff::SignedDuration` type that is effectively identical to `std::time::Duration` (including its floating point APIs), but is signed. And then there will be APIs for converting from a `SignedDuration` to a `Span`.

This still isn't quite as convenient as building a `Span` directly from `f64` of seconds. But I think a direct APIs is probably a bit more involved. Ideally one would be able to specify the whole units of the floating point (like "convert `2.5 years` to a `Span`"), and I think that will require more design work. But at least the conversion through a `SignedDuration` will do everything correctly and absolve you of having to do math that I agree looks pretty annoying.

Reply:
> If you know your `seconds_f` is always positive, you can create a `std::time::Duration` from it and then use `Span::try_from(duration)`.

`seconds_f` can definitely be negative here. I *could* take the absolute value of `seconds_f` and keep track of the sign myself, swapping `checked_add`/`checked_sub` below in case it was negative. But that's also not really better than what we have now.

> If it's possibly negative, my plan is to add a `jiff::SignedDuration` type that is effectively identical to `std::time::Duration` (including its floating point APIs), but is signed. And then there will be APIs for converting from a `SignedDuration` to a `Span`.

That would work for us. But the current version is also completely fine.

Reply:
Aye. I plan to add `SignedDuration` for other reasons as well, but I think a side effect is indeed that this should be a little nicer.